src/audio/copier/copier.c

```diff
@@ -571,6 +544,39 @@ static int copier_copy(struct comp_dev *dev)
 	return ret;
 }
 
+/* Copier has one input and one or more outputs. Maximum of one gateway can be connected
+ * to copier or no gateway connected at all. Gateway can only be connected to either input
+ * pin 0 (the only input) or output pin 0. With or without connected gateway it is also
+ * possible to have component(s) connected on input and/or output pins.
+ *
+ * A special exception is a multichannel ALH gateway case. These are multiple gateways
+ * but should be treated like a single gateway to satisfy rules above. Data from such
+ * gateways has to be multiplexed into single stream (for input gateways) or demultiplexed
+ * from single stream (for output gateways) so such gateways work kind of like a single
+ * gateway, i.e., produce/consume single stream.
+ */
+static int copier_copy(struct comp_dev *dev)
+{
+	struct copier_data *cd = comp_get_drvdata(dev);
+
+	comp_dbg(dev, "copier_copy()");
+
+	switch (dev->ipc_config.type) {
+	case SOF_COMP_HOST:
+		if (!cd->ipc_gtw)
+			return host_common_copy(cd->hd, dev, copier_host_dma_cb);
+		return 0;
+	case SOF_COMP_DAI:
+		if (cd->endpoint_num == 1)
+			return dai_common_copy(cd->dd[0], dev, cd->converter);
+		break;
+	default:
+		break;
+	}
+	/* handle multi-endpoint and maybe type == none copy */
+	return do_multi_endpoint_copy(cd, dev);
```

Comment:
this is a misnomer. This case handles both the multi-endpoint case for DAI's and the copy for module copiers.

Reply:
I agree on this, when I tried to change for this, I found it is not only for multiendpoint, so there are two ways:

1. simply change the function name, like: do_multi_endpoint_module_copy
2. split this function further to two, do_multi_endpoint_copy and do_module_copy

I like 2, what's your opinion?